[Pin Sern] Duke Increments

fangpinsern opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

fangpinsern committed:
Level1GreetEchoExit

fangpinsern committed:
Level1GreetEchoExit2.0

fangpinsern committed:
Level2AddList

fangpinsern committed:
Level3MarkAsDone

fangpinsern committed:
Level4ToDosEventsDeadlines

fangpinsern committed:
Level5HandleErrors

fangpinsern committed:
Level6Delete
Delete errors not handled yet

fangpinsern committed:
Level 6
version 2.0
Delete error still not done.

fangpinsern committed:
Level 7
1.0
added file read and write capabilities

fangpinsern committed:
Level 8
1.0
improved date functionality

fangpinsern committed:
Level 7
File Reading and writing capabilities included

fangpinsern committed:
Level-7
Read and Write to file capabilities added

fangpinsern committed:
Level 7
Read and Write to file capabilities added

fangpinsern committed:
Level 8
Time and date

Merge branch 'branch-Level-8'

fangpinsern committed:
A-MoreOOP

abstract classes out

fangpinsern committed:
A-Packages

grouped tasks into a task package

fangpinsern committed:
Packaging 2.0

more packaging

fangpinsern committed:
JUnit Testing files

Added JUnit testing files to the src folder

fangpinsern committed:
JUnit Testing 2.0

Previously no code was in the file

fangpinsern committed:
Addition of JavaDoc

Javadoc added to the classes

fangpinsern committed:
JavaDoc 3.0

This time with branching

fangpinsern committed:
JavaDoc 4.0

Added new JavaDoc in branch

fangpinsern committed:
Level 9 - Find Functionality added to Duke

fangpinsern committed:
Level 9 2.0 - Find function improved

Message will show empty list when there is nothing in the list

fangpinsern committed:
Merge branch 'Level-9'

fangpinsern committed:
Merge branch 'gradle'

fangpinsern committed:
A-Gradle

Gradle capability added

fangpinsern committed:
Check Style

stevenwjy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,246 @@
+        //     storage.printFileContents(li);
+        // } catch(FileNotFoundException e){
+        //     //System.out.println("File not found");
+        // }
```
It would be better if you could just remove these unused part of the code instead of just commenting it.

stevenwjy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,246 @@
+            if (firstWord.equals("bye")) {
+                // if bye, then end the program
+                ui.printline();
+                System.out.println("\tBye. Hope to see you again soon!");
```
Since you already have a `UI` class, you may want to use that class to handle all interactions with user (i.e. handle all input and output). It might be a bit confusing if sometimes you print from `UI` class and sometimes directly from `Duke`.

stevenwjy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,246 @@
+                if (taskNumb >= tasks.size()) {
+                    error = "taskDoNotExist";
+                } else if (tasks.getTask(taskNumb).getIsDone()) {
+                    //System.out.println("im here");
```
It would be to not commit code that is used only for debugging.

stevenwjy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,246 @@
+import duke.command.Parser;
+import duke.command.Storage;
+import duke.command.TaskList;
+import duke.command.Ui;
```
You may want to consider creating another separate package for `Ui` and moving `TaskList` class  to `duke.tasks` package.

stevenwjy commented on src/main/java/duke/command/AException.java:
```diff
@@ -0,0 +1,66 @@
+ * Represents every exception that Duke may encounter
+ * when the user uses Duke.
+ */
+public class AException {
```
You may consider passing a `String` that describes the error when you construct an `AException` and create an object for every exception that you encounter. Hence, every error is described by one exception object instead of just calling a function that directly prints to stdout.

stevenwjy commented on src/main/java/duke/command/Ui.java:
```diff
@@ -0,0 +1,106 @@
+     * @param t task that is being added into the list
+     * @param i number of task that are in the list
+     */
+    public void takeInput(Task t, int i) {
```
It would be better if you can give a more descriptive name.

stevenwjy reviewed (commented):
Overall the code looks nice and it seems like you have put a lot of efforts into it, great job!!

However, here are some feedback that you may want to consider in order to improve the code quality:
- Add newlines at the end of files to make it look nicer when you push to GitHub (prevent it from showing a read symbol that indicates no newline at the end of a file).
- Avoid mixing all the logic in one place (i.e.`run` function inside `Duke` class), it might be better to create a separate class for each command that has its own execution function.
- Don't put too many responsibilities for your Duke class (i.e. handle input/output, parsing different types of command, and handle command execution), you may want to break those functionalities into multiple classes.
- It would be better to have a class that handles all input/output. This will also help you in a debugging process.
- It would be better if you ignore the .class files when you push your code to GitHub.

fangpinsern committed:
Level 10 JavaFX

JavaFX GUI inserted
Logic incomplete

fangpinsern committed:
Level - 10 Improved

GUI logic improved.
Minor bug fixes

fangpinsern committed:
Assersions added

fangpinsern committed:
A-CodeQuality

KartikeKhandelwal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,362 @@
+    /**
+     * Creates a Duke object that takes in a filepath to
+     * the file where the list of previous tasks is stored.
+     * @param filePath path to where the last saved task list is stored
```
Leave line here

KartikeKhandelwal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,362 @@
+     */
+    public String run(String input) {
+        try {
+//        tasks = new TaskList(new LinkedList<Task>());
```
remove unused code

KartikeKhandelwal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,362 @@
+    
+    
+//    public static void main(String[] args) {
+//        new Duke("duke.txt");
```
again, remove

KartikeKhandelwal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,362 @@
+        stage.setScene(scene);
+        stage.show();
     }
 }
```
For run function, need to add abstract out the parsing to your parser class, it's too long.  Too many newline comments. 
Abstract out error handling, do not add to the main.

KartikeKhandelwal commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,362 @@
+    /**
+     * Runs the logic for Duke.
+     */
+    public String run(String input) {
```
run itself can be split into multiple functions within the class

KartikeKhandelwal commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+     * @param doner the completion status of the task
+     *              (Completed is 1, incomplete is 2)
+     */
+    public Deadline(String description, Date by, int doner) {
```
doner can be renamed to isDone
and if the status can only be 1/2, change to Boolean type.

KartikeKhandelwal commented on src/main/java/duke/utils/AException.java:
```diff
@@ -0,0 +1,66 @@
+package duke.utils;
```
should be a subclass of exception. 
for multiple exception, should subclass DukeException class
Rename the class to more meaningful names such as DukeException

KartikeKhandelwal reviewed (commented):
Needs improvement in structure and abstraction.

fangpinsern committed:
Removed Unnecessary Code

Removed all previous code comments that I could not let go

fangpinsern committed:
C-DetectDuplicates

Detection of duplicate Descriptions
Will not add the task if duplicated

fangpinsern committed:
Assert Command added

Assert that firstWord is not null

fangpinsern committed:
Merge pull request #5 from fangpinsern/Assertions

Assert Command added

fangpinsern committed:
Improve certain code styling

fangpinsern committed:
Merge pull request #6 from fangpinsern/CodeStyling

Improve certain code styling

fangpinsern committed:
Minor bug fixes

fangpinsern committed:
Minor Bug Fixes

fangpinsern committed:
Merge branch 'CodeStyling'

fangpinsern committed:
Readme.md added

Readme updated to include features and usage examples
Minor bug fixes
Improved visuals

fangpinsern committed:
Merge branch 'master' of https://github.com/fangpinsern/duke

fangpinsern committed:
Set theme jekyll-theme-midnight

fangpinsern committed:
Added Ui.png

Ui.png added to docs folder
Minor updates to README.md
Minor bug fixes

fangpinsern committed:
Merge branch 'master' of https://github.com/fangpinsern/duke

fangpinsern committed:
Set theme jekyll-theme-slate

fangpinsern committed:
Set theme jekyll-theme-slate

fangpinsern committed:
Changes to README.md

fangpinsern committed:
Merge branch 'master' of https://github.com/fangpinsern/duke

fangpinsern committed:
Minor fixes to README.md

fangpinsern committed:
Changed _config.yml

Changed title and description of the page

fangpinsern committed:
JAR File added

JAR File in build/libs

fangpinsern committed:
Minor Bug Fixes

Minor bug fixes in manifest and gradle build
